.github/workflows/docker-publish.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and Publish image to Docker Hub
+on:
+    push:
+      branches:
+        - 'master'
+      tags:
+        - 'v*.*.*'
+
+jobs:
+  build_and_push:
+    env:
+      DOCKER_HUB: ${{ secrets.DOCKER_LOGIN }}
+      DOCKER_TAGS: type=ref,event=branch,branch=master,latest=true
+    runs-on: ubuntu-22.04
+    permissions:
+      id-token: write
+      attestations: write
+      contents: read
+      packages: write
+
+    strategy:
+      matrix:
+        target:
+          - name: stone-prover
+            dockerfile: Dockerfile
+          - name: cpu_air_prover
+            dockerfile: air_prover/Dockerfile
+          - name: cpu_air_verifier
+            dockerfile: air_verifier/Dockerfile
+    continue-on-error: true
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKER_LOGIN }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Determine Docker Tags
+        id: set-tag
+        run: |
+          if [[ "${GITHUB_REF}" == refs/tags/* ]]; then
+            TAG_NAME=$(echo ${GITHUB_REF} | sed 's/refs\/tags\///')
+            echo "DOCKER_TAGS=${TAG_NAME}" >> $GITHUB_ENV
+          elif [[ "${GITHUB_REF}" == "refs/heads/master" ]]; then
+            echo "Latest version tags..."
```

Comment:
Great! May I ask you to trigger on-push-master to test it?

Reply:
Is there any problem with doing it on master? Or can I do it on another branch for testing?

Reply:
Done @dmirgaleev, please check.

https://github.com/jsanchez556/stone-packaging/actions/runs/11153195071/